ForumProject/communications/consumers.py

```diff
@@ -0,0 +1,110 @@
+import json
+
+from asgiref.sync import async_to_sync
+from channels.generic.websocket import WebsocketConsumer
+
+from .models import Room, Message
+
+
+class ChatConsumer(WebsocketConsumer):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(args, kwargs)
+        self.room_name = None
+        self.room_group_name = None
+        self.room = None
+        self.messages = None
+        self.user = None
+        self.user_inbox = None
+
+    def connect(self):
+        self.room_name = self.scope['url_route']['kwargs']['room_name']
+        self.room_group_name = f'chat_{self.room_name}'
+        self.room = Room.objects.get(name=self.room_name)
```

Comment:
The line might raise a `Room.DoesNotExist` exception if the room name does not exist in the database. Consider handling this exception to either create the room or send an error message back to the client.

Reply:
Done, I added try-except and logging for this case